### PR TITLE
Diego Romero challenge inject

### PR DESCRIPTION
**Pull request by Diego-Romero**

### CODE REVIEW

**Comment by Diego-Romero** [comment]
diego Romero pull request

File: lib/Array.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+		self.each do |element| 
+			array = yield array,element
+		end
+		p array
+	end
+
+
+end
+
+
```
Comment (dbugsy) [inline lib/Array.rb:14]: Indentation - change the preferences in Sublime to ensure you are using 2 spaces instead of tabs.

Comment (dbugsy) [inline lib/Array.rb:14]: don't use puts or p

File: spec/inject_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	it 'can deduct each of its own elements' do
+		expect([1,2,3,4].update_with(0){|array, element| array-element}).to eq -10
+	end
+
+	it 'can divide each of its own elements' do
+		expect([30,60].update_with(1){|array, element| array/element}).to eq 2
+	end
+
+
+end
```
Comment (dbugsy) [inline spec/inject_spec.rb:21]: Make sure that you are comparing to real inject.

i.e.

```
it 'can update each of its elements' do
  expect([1,2,3,4].update_with(1) {|array, element| array * element}).to eq [1,2,3,4].inject(1) {|array, element| array * element}
end
```

Comment (dbugsy) [inline spec/inject_spec.rb:21]: Check all the examples in the Ruby docs - can you find the longest word? Can your update_with method handle a symbol like inject can?

File: lib/Array.rb (hunk 1 above)
Comment (dbugsy) [inline lib/Array.rb:14]: check your naming - does update_with really describe what this method does?

**Comment by spike01** [comment]
:checkered_flag:

**Comment by spike01** [comment]
:checkered_flag:

**Pull request closed without merging**